docs/auth/oidc.md

```diff
@@ -0,0 +1,235 @@
+---
+id: oidc
+title: OIDC provider from scratch
+description: This section shows how to use an OIDC provider from scrath, same steps apply for custom providers.
+---
+
+This section shows how to use an OIDC provider from scratch, same steps apply for custom
+providers. Please note these steps are for using a provider, not how to implement one.
+
+## Summary
+
+To add providers not enabled by default like OIDC, we need to follow some steps, we
+assume you already have a sign in page to which we'll add the provider so users can
+sign in through the provider. In simple steps here's how you enable the provider:
+
+- Create an API reference to identify the provider.
+- Create the API factory that will handle the authentication.
+- Add a resolver so you can handle the result from the authentication.
+- Configure the provider to access your 3rd party auth solution.
+- Add the provider to sign in page so users can login with it.
+
+We'll explain each step more in detail next.
+
+### The API reference
+
+An API reference exist for the sake of **Dependency Injection**, it's basically an ID to
```

Comment:
I'm unsure if this guide should explain the dependency injection aspects. Might be better to instead link to https://backstage.io/docs/api/utility-apis and keep this guide as slim as possible to reduce the amount of text.

Reply:
Shortened and linked that url, thanks.